File: src/models/BrandKit.js
const mongoose = require('mongoose');

const brandKitSchema = new mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
        unique: true // One Brand Kit per user (for now)
    },
    name: {
        type: String,
        default: 'My Brand'
    },
    voiceDescription: {
        type: String,
        // e.g., "Professional, confident, and witty. Uses emojis sparingly."
        default: ''
    },
    website: {
        type: String,
        default: ''
    },
    // Brand Colors (Hex codes)
    colors: [{
        type: String,
        match: /^#([0-9a-f]{3}){1,2}$/i
    }],
    // Reference documents for RAG (Retrieval Augmented Generation)
    // Simple array of URLs for now
    documents: [{
        name: String,
        url: String,
        type: String // 'pdf', 'txt', 'md'
    }]
}, {
    timestamps: true
});

module.exports = mongoose.model('BrandKit', brandKitSchema);

File: app/models/brand_kit.py
from datetime import datetime
from typing import List, Optional
from beanie import Document, Link
from pydantic import Field, BaseModel
from .user import User

class BrandDocument(BaseModel):
    name: str
    url: str
    doc_type: str = Field(alias="type")

class BrandKit(Document):
    user_id: Link[User] = Field(alias="userId")
    name: str = "My Brand"
    voice_description: str = Field("", alias="voiceDescription")
    website: str = ""
    colors: List[str] = Field(default=[])
    documents: List[BrandDocument] = Field(default=[])
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "brand_kits"
        indexes = ["userId"]
